threefoldtech/tfgrid-sdk-ts
# Feat: suppot TFGrid-KYC

### Description

Add KYC client to grid client, dashboard, and types packages.


changes: 

- grid client:
   - add kyc client
   - add kyc client to grid client
   - check if kyc verified on create name and handle methods in twinddeployment 
   
- types:  add related error classes 

- dashboard:
   - add kyc verifier that uses iframe to connect to idenfy
   - add kyc store, init on profile, clear if no profile
   - prevent deployment if kyc is not verified, in view layout
 
### tested scenarios:
 - try to deploy without kyc verification all deployments should be blocked
 - check twin page to see the kyc status
 - use kyc client and check its methods
 - check error handling in dashboard, toast etc, you can manipulate the headers in kyc client
 - check if you can deploy without kyc in gridclient
 - try to enter the verification process then cancel 
 - try to have rejected kyc verification
 - try to verify the kyc with valid data using the dashboard
 - try to scan the qr code  and continue verification process on mobile 
- make sure that the kyc status got cleaned after logout/refresh

related issues: 

- #3519 

info issue: https://github.com/threefoldtech/info_grid/issues/666

screenshots: 
![Screenshot from 2024-10-23 12-21-41](https://github.com/user-attachments/assets/13c7730e-fd85-4c30-a79d-17eee4df90a0)
![Screenshot from 2024-10-23 12-21-33](https://github.com/user-attachments/assets/1d3ce4cb-d64d-4746-a779-99b248713230)
![image](https://github.com/user-attachments/assets/a3af7d72-be03-4353-b9ac-934010d9a12d)
![Screenshot from 2024-10-23 12-27-40](https://github.com/user-attachments/assets/dba0d698-43c8-4bb9-a701-010221cb8b75)
![Screenshot from 2024-10-23 12-27-19](https://github.com/user-attachments/assets/236dee09-18c8-4e3e-8228-6a4162928c62)
![image](https://github.com/user-attachments/assets/6cf3bbcb-ac41-4bdc-b43c-eb21fac00b3c)
![image](https://github.com/user-attachments/assets/b13e125b-6347-4842-b62e-2f8aeb8c4bd1)
